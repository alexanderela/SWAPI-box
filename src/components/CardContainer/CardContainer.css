.CardContainer {
  border: 1px dotted purple;
  display: grid;
	grid-template-columns: repeat( auto-fit, minmax(15rem, 1fr) );
  height: 100vh;
  background: gray;
  margin: 3rem 5rem;
}

.button-container {
  display: flex;
  justify-content: space-around;
  width: 100%;
  height: 5rem;
}