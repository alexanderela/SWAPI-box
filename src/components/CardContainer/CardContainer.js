import React, { Component } from 'react';
import './CardContainer.css';
import Card from '../Card/Card';

class CardContainer extends Component {
	constructor(props) {
		super(props);
		this.state = {

		}
	}

	render() {
		return (
			<div className='CardContainer'>
				<div className="button-container">
					<button className="cat-button people-button">People</button>
					<button className="cat-button planets-button">Planets</button>
					<button className="cat-button vehicles-button">Vehicles</button>
				</div>
				<Card />
				<Card />
				<Card />
				<Card />
			</div>
		);
	}
}

export default CardContainer;