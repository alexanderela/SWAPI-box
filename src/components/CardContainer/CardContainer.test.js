import React from 'react';
import ReactDOM from 'react-dom';
import CardContainer from './CardContainer';
import { shallow } from 'enzyme';

it('matches the snapshot', () => {
	const wrapper = shallow(<CardContainer />);
	expect(wrapper).toMatchSnapshot();
})