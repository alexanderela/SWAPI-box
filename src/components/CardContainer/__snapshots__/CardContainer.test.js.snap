// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardContainer />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="button-container"
        >
          <button
            className="cat-button people-button"
          >
            People
          </button>
          <button
            className="cat-button planets-button"
          >
            Planets
          </button>
          <button
            className="cat-button vehicles-button"
          >
            Vehicles
          </button>
        </div>,
        <Card />,
        <Card />,
        <Card />,
        <Card />,
      ],
      "className": "CardContainer",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <button
              className="cat-button people-button"
            >
              People
            </button>,
            <button
              className="cat-button planets-button"
            >
              Planets
            </button>,
            <button
              className="cat-button vehicles-button"
            >
              Vehicles
            </button>,
          ],
          "className": "button-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "People",
              "className": "cat-button people-button",
            },
            "ref": null,
            "rendered": "People",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Planets",
              "className": "cat-button planets-button",
            },
            "ref": null,
            "rendered": "Planets",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Vehicles",
              "className": "cat-button vehicles-button",
            },
            "ref": null,
            "rendered": "Vehicles",
            "type": "button",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="button-container"
          >
            <button
              className="cat-button people-button"
            >
              People
            </button>
            <button
              className="cat-button planets-button"
            >
              Planets
            </button>
            <button
              className="cat-button vehicles-button"
            >
              Vehicles
            </button>
          </div>,
          <Card />,
          <Card />,
          <Card />,
          <Card />,
        ],
        "className": "CardContainer",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <button
                className="cat-button people-button"
              >
                People
              </button>,
              <button
                className="cat-button planets-button"
              >
                Planets
              </button>,
              <button
                className="cat-button vehicles-button"
              >
                Vehicles
              </button>,
            ],
            "className": "button-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "People",
                "className": "cat-button people-button",
              },
              "ref": null,
              "rendered": "People",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Planets",
                "className": "cat-button planets-button",
              },
              "ref": null,
              "rendered": "Planets",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Vehicles",
                "className": "cat-button vehicles-button",
              },
              "ref": null,
              "rendered": "Vehicles",
              "type": "button",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
